processor/transformprocessor/internal/metrics/func_scale.go

```diff
@@ -0,0 +1,130 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlmetric"
+)
+
+type ScaleArguments struct {
+	Multiplier float64
+	Unit       ottl.Optional[ottl.StringGetter[ottlmetric.TransformContext]]
+}
+
+func newScaleMetricFactory() ottl.Factory[ottlmetric.TransformContext] {
+	return ottl.NewFactory("scale_metric", &ScaleArguments{}, createScaleFunction)
+}
+
+func createScaleFunction(_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[ottlmetric.TransformContext], error) {
+	args, ok := oArgs.(*ScaleArguments)
+
+	if !ok {
+		return nil, fmt.Errorf("ScaleFactory args must be of type *ScaleArguments[K]")
+	}
+
+	return Scale(*args)
+}
+
+func Scale(args ScaleArguments) (ottl.ExprFunc[ottlmetric.TransformContext], error) {
+	return func(ctx context.Context, tCtx ottlmetric.TransformContext) (any, error) {
+		metric := tCtx.GetMetric()
+
+		var unit string
```

Comment:
lets make unit a `*string` and then bellow instead of having to do `!args.Unit.IsEmpty()` again we can do `unit != nil`

Reply:
done